package.json

```diff
@@ -28,6 +28,8 @@
     "build:mac": "yarn run compile && electron-builder --mac --dir",
     "build:win": "yarn run compile && electron-builder --win --dir",
     "integration": "jest --runInBand --detectOpenHandles --forceExit integration",
+    "test:unit": "jest --watch --testPathIgnorePatterns integration",
+    "test:integration": "func() { jest ${1:-xyz} --watch --runInBand --detectOpenHandles --forceExit --modulePaths=[\"<rootDir>/integration/\"]; }; func",
```

Comment:
What file changes will cause tests to rerun? I edited/saved some files but no tests were run ("No tests found, exiting with code 0").

Reply:
For `unit tests`, watch triggers after a change in most production or test code (and also some configuration files). The initial behavior of `jest --watch` is to use git to find uncommitted production and test-code, and run all tests directly or indirectly affected by those changes. 

In your case: 
1. The changed files need to be related to a test for anything to happen.
2. Our current unrelated configuration expects `*.test`-files to contain at least one test in order to not be perceived as failure. This explains `"No tests found, exiting with code 0"`.

Deviating from this, for `integration tests` the watch triggers only for changes in test code, as the tests happen over already built application. Still, this makes tweaking the slow integration tests a bit more friendly :)

FYI: In watch mode, you can also use `w` to see additional controls. Eg. `p` or `t` will make you able to specify explicitly which tests to run, `a` will make you run all tests despite git, and `f` will make you re-run only failing tests. Cool stuff, right? :)

Reply:
pretty sure I edited an integration test file (edited some string, maybe the test name). I'll try again

Reply:
I added a test to `integration/__tests__/cluster-pages.tests.ts` and saved. No tests found. Pressing `a` runs the tests. Is this how to use it?

Reply:
We could have a sesh? Or all this should be documented somewhere within jest. Hmm.